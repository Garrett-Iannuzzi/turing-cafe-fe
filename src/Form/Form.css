form {
  border: solid 3px black;
  background-color: white;
  display: flex;
  height: 13em;
  flex-direction: column;
  padding: 1em;
  width: 35%
}

.make-res-btn {
  cursor: pointer;
  margin-top: .5em;
}