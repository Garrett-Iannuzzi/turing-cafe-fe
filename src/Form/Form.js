import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor(props) {
    super(props)
    this.state ={
      name: '',
      date: '',
      time: '',
      number: 0
    }
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  resetInputs = () => {
    this.setState({ name: '', date: '', time: '', number: 0 })
  }

  submitReservation = (e) => {
    e.preventDefault()
    const newRes = { id: Date.now(), ...this.state }
    this.props.handleAddReservation(newRes)
    this.resetInputs()
  }

  render() {
    return(
      <form>
        <label>Name:</label>
        <input 
          className='input-1'
          name='name'
          value={this.state.name}
          onChange={ (e) => this.handleChange(e) }
        />
        <label>Date:</label>
        <input
          className='input-2'
          name='date'
          value={this.state.date}
          placeholder='mm/dd'
          onChange={ (e) => this.handleChange(e) } 
        />
        <label>Time:</label>
        <input
          className='input-3'          
          name='time'
          value={this.state.time}
          onChange={ (e) => this.handleChange(e) } 
        />
        <label>Number of Guests:</label>
        <input
          className='input-4'
          name='number'
          value={this.state.number}
          onChange={ (e) => this.handleChange(e) } 
        />
        <button className='make-res-btn' onClick={ (e) => this.submitReservation(e) }>Make Res!</button>
      </form>
    )
  }
}

export default Form;