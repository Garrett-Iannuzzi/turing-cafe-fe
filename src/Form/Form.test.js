import React from 'react';
import { shallow } from 'enzyme';
import Form from '../Form/Form';

describe('Form', () => {

  it('Should match the snapshot with the correct data', () => {
    const wrapper = shallow(<Form />)

    expect(wrapper).toMatchSnapshot()
  });

  it('Should update state when handleChange is called', () => {
    const wrapper = shallow(<Form />)
    const mockEvent = { target: { name: 'name', value: 'Nick' } }
    wrapper.setState({ name: '', date: '', time: '', number: 0 });
    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state()).toEqual({ name: 'Nick', date: '', time: '', number: 0 })
  });
});