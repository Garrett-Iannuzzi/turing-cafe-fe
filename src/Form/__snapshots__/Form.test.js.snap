// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Should match the snapshot with the correct data 1`] = `
<form>
  <label>
    Name:
  </label>
  <input
    className="input-1"
    name="name"
    onChange={[Function]}
    value=""
  />
  <label>
    Date:
  </label>
  <input
    className="input-2"
    name="date"
    onChange={[Function]}
    placeholder="mm/dd"
    value=""
  />
  <label>
    Time:
  </label>
  <input
    className="input-3"
    name="time"
    onChange={[Function]}
    value=""
  />
  <label>
    Number of Guests:
  </label>
  <input
    className="input-4"
    name="number"
    onChange={[Function]}
    value={0}
  />
  <button
    className="make-res-btn"
    onClick={[Function]}
  >
    Make Res!
  </button>
</form>
`;

exports[`ReservationCard Should match the snapshot with the correct data 1`] = `
<article>
  <h1
    className="article-h1"
  >
    Name: 
  </h1>
  <h2
    className="article-h2"
  >
    Date: 
  </h2>
  <h3
    className="article-h3"
  >
    Time: 
  </h3>
  <h3
    className="article-h4"
  >
    Number of guests: 
  </h3>
</article>
`;
